File: Ejercicios/JavaScript/Interfaces/CalculadoraBasica.js
class CalculadoraBasica {
  constructor(nombre, version) {
    if (nombre === undefined && version === undefined) {
      this.nombre = "Básica";
    } else if (nombre instanceof CalculadoraBasica) {
      this.nombre = nombre.nombre;
    } else if (version === undefined) {
      this.nombre = nombre;
    } else {
      this.nombre = `${nombre} v${version}`;
    }
  }

  sumar(a, b) {
    return a + b;
  }
}

export default CalculadoraBasica;

File: Ejercicios/Java/Interfaces/CalculadoraBasica.java
interface ICalculadora {
    int sumar(int a, int b);
}

public class CalculadoraBasica implements ICalculadora {
    private String nombre;

    public CalculadoraBasica() {
        this("Básica");
    }

    public CalculadoraBasica(String nombre) {
        this.nombre = nombre;
    }

    public CalculadoraBasica(String nombre, String version) {
        this.nombre = nombre + " v" + version;
    }

    public CalculadoraBasica(CalculadoraBasica otra) {
        this.nombre = otra.nombre;
    }

    @Override
    public int sumar(int a, int b) {
        return a + b;
    }
}
